File: p026-050/p28.c
// Project Euler
// Alex Johnson
// Problem 28: Number spiral diagonals

#include <stdio.h>

int main(void) {

  int spiral_width = 1001;
  int ring_num = spiral_width / 2;

  int sum = 1;

  int i;
  for (i = 1; i <= ring_num; i++) {
    int ring_width = i * 2 + 1;
    int ring_square = ring_width * ring_width;
    int step = ring_width - 1;

    int j;
    for (j = 0; j < 4; j++) {
      sum += ring_square;
      ring_square -= step;
    }

  }

  printf("%d\n", sum);
  return 0;
}

File: p026-050/p28.py
# Project Euler
# Alex Johnson
# Problem 28: Number spiral diagonals

spiral_width = 1001
ring_num = spiral_width // 2

total = 1
for i in range(2, ring_num + 1):
    ring_width = i * 2 + 1
    ring_square = ring_width ** 2
    step = ring_width - 1
    for j in range(4):
        total += ring_square
        ring_square -= step

print(total)
